### PR TITLE
initial working version

### PR DESCRIPTION
**Pull request by charlypoly**
cf: https://github.com/dotansimha/graphql-code-generator/issues/8409

### CODE REVIEW

File: src/fixtures/validFile.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type Config = { test?: boolean };
```
Comment (dotansimha) [inline src/fixtures/validFile.ts:1]: maybe also add a fixture for a valid file that uses non-default?

File: src/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { readFileSync } from "fs";
+import { resolve } from "path";
+import { TypeScriptSWCLoader } from "./index";
+
+describe("TypeScriptSWCLoader", () => {
```
Comment (dotansimha) [inline src/index.spec.ts:5]: maybe we should also have a test suite that uses cosmiconfig package and test the loader with it?

File: src/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const fixturesPath = resolve(__dirname, "fixtures");
+
+  describe("loading a valid configuration typescript file", () => {
+    describe("without options", () => {
+      it("should return a valid config object", () => {
+        expect(
+          TypeScriptSWCLoader()(
+            "",
+            readFileSync(resolve(fixturesPath, "validFile.ts")).toString()
```
Comment (B2o5T) [inline src/index.spec.ts:14]:
```suggestion
            readFileSync(resolve(fixturesPath, "validFile.ts"), 'utf8')
```

File: src/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    });
+  });
+
+  describe("loading an INVALID configuration typescript file", () => {
+    describe("without options", () => {
+      it("should throw an error", () => {
+        expect(() =>
+          TypeScriptSWCLoader()(
+            "",
+            readFileSync(resolve(fixturesPath, "invalidFile.ts")).toString()
```
Comment (B2o5T) [inline src/index.spec.ts:29]:
```suggestion
            readFileSync(resolve(fixturesPath, "invalidFile.ts"), 'utf8')
```

**Review by dotansimha — CHANGES_REQUESTED** [review]

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,6 @@
+node_modules/
```
Comment (B2o5T) [inline .prettierignore:1]: prettier ignore by default node_modules
```suggestion
```

File: .prettierignore [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+dist/
+yarn.lock
```
Comment (B2o5T) [inline .prettierignore:3]:
```suggestion
```

prettier don't format lock files

File: .eslintignore [hunk 7]
```diff
@@ -0,0 +1,5 @@
+node_modules/
```
Comment (B2o5T) [inline .eslintignore:1]:
```suggestion
```

File: .eslintignore [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+dist/
+yarn.lock
```
Comment (B2o5T) [inline .eslintignore:3]:
```suggestion
```

File: .prettierignore [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+.bob/
+.github/
```
Comment (B2o5T) [inline .prettierignore:5]:
```suggestion
```
`yaml` files still can be formatted by prettier

File: bob.config.cjs [hunk 10]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (B2o5T) [inline bob.config.cjs:1]:
```suggestion
```
I guess this file can be removed?

File: package.json [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  "author": "Charly POLY <cpoly55@gmail.com>",
+  "license": "MIT",
+  "scripts": {
+    "test": "jest --passWithNoTests",
+    "clean": "rimraf dist",
+    "prebuild": "yarn clean",
+    "build": "bob build --single",
+    "prettier": "prettier --write --list-different .",
+    "prettier:check": "prettier --check .",
+    "lint": "eslint .",
```
Comment (B2o5T) [inline package.json:16]:
```suggestion
    "lint": "eslint --ignore-path .gitignore .",
```
and `.eslintignore` can be removed

File: .github/workflows/main.yml [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    name: Unit Test on Node ${{matrix.node_version}} (${{matrix.os}})
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - prettier-check
+      - esm
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        node_version: [12, 14, 16, 18]
```
Comment (B2o5T) [inline .github/workflows/main.yml:60]: do we need tests for 12?
```suggestion
        node_version: [14, 16, 18]
```

File: .github/workflows/main.yml [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  prettier-check:
+    name: 🧹 Prettier Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Master
+        uses: actions/checkout@v3
+      - name: Setup env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 17
```
Comment (B2o5T) [inline .github/workflows/main.yml:31]:
```suggestion
          nodeVersion: 18
```
as 18 is lts

File: .github/workflows/main.yml [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+  esm:
+    name: Testing exports integrity
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 17
```
Comment (B2o5T) [inline .github/workflows/main.yml:43]:
```suggestion
          nodeVersion: 18
```

File: src/index.spec.ts (hunk 2 above)
Comment (charlypoly) [reply src/index.spec.ts:6]: done: ea930c9

File: src/fixtures/validFile.ts (hunk 1 above)
Comment (charlypoly) [reply src/fixtures/validFile.ts:1]: done: ea930c9

**Pull request merged**